### PR TITLE
Np 399 update datamodel

### PR DESCRIPTION
**Pull request by janbj**

### CODE REVIEW

File: src/main/java/no/unit/nva/model/Contributor.java [hunk 1]
```diff
@@ -1,17 +1,14 @@
 package no.unit.nva.model;
 
-import java.net.URI;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
```
[the remark this answers is not in the document]
Comment (janbj) [reply src/main/java/no/unit/nva/model/Contributor.java]: Orestis said no

File: src/main/java/no/unit/nva/model/Contributor.java [hunk 2]
```diff
@@ -20,53 +17,26 @@ public Contributor() {
...
 
-    public void setNameType(String nameType) {
-        this.nameType = nameType;
+    public void setIdentity(Identity identity) {
+        this.identity = identity;
     }
 
-    public List<String> getImportedAffiliation() {
-        return importedAffiliation;
+    public List<Organization> getAffiliation() {
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/model/Contributor.java]: If affiliations are always plural, we should make this explicit in the nomenclature

File: src/main/java/no/unit/nva/model/EntityDescription.java [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
+
+import java.net.URI;
 import java.util.List;
 import java.util.Map;
 
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
 public class EntityDescription {
 
-    private String type;
+    private String publicationType;
```
Comment (brinxmat) [inline src/main/java/no/unit/nva/model/EntityDescription.java]: I suspect publication type is not a string, there are a limited set of possibilities, we should speak with PO, for now JournalArticle is the only type.

File: src/main/resources/publicationContext.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
+{
+  "@vocab": "https://example.org/cheeky#",
```
Comment (brinxmat) [inline src/main/resources/publicationContext.json]: Probably not.

File: src/main/resources/publicationFrame.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{
+  "@context": {
+    "@vocab": "https://example.org/cheeky#",
```
Comment (brinxmat) [inline src/main/resources/publicationFrame.json]: Again :)

File: src/test/java/no/unit/nva/model/util/ContextUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+public class ContextUtilTest {
+
+    @Test
+    public void testInjectContextIntoJsonNodeNotAnObject() {
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        ArrayNode arrayNode = objectMapper.createArrayNode();
+        ObjectNode objectNode = objectMapper.createObjectNode();
+
+        Assertions.assertThrows(IllegalArgumentException.class, () -> {
```
Comment (brinxmat) [inline src/test/java/no/unit/nva/model/util/ContextUtilTest.java]: Do you want to test the message that is returned?

File: src/main/resources/publicationContext.json (hunk 4 above)
Comment (janbj) [reply src/main/resources/publicationContext.json]: What should we use then?

File: src/main/java/no/unit/nva/model/EntityDescription.java (hunk 3 above)
Comment (janbj) [reply src/main/java/no/unit/nva/model/EntityDescription.java]: Yes, I tried using the enum but I couldn't guarantee all possible mappings when Terje was testing various DOIs

File: src/main/resources/publicationFrame.json (hunk 5 above)
Comment (janbj) [reply src/main/resources/publicationFrame.json]: Sea otter

File: src/main/java/no/unit/nva/model/EntityDescription.java (hunk 3 above)
Comment (brinxmat) [reply src/main/java/no/unit/nva/model/EntityDescription.java]: This got me to thinking about datacite and crossref, we should talk with JEG

**Pull request merged**